src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.wasm.cs

```diff
@@ -119,6 +98,17 @@ public static void SetBorder(UIElement element, Thickness thickness, Brush brush
 						break;
 				}
 			}
+
+			if (cornerRadius == CornerRadius.None)
+			{
+				element.ResetStyle("border-radius", "overflow");
+			}
+			else
+			{
+				var outer = cornerRadius.GetRadii(element.RenderSize, thickness).Outer;
+				WindowManagerInterop.SetCornerRadius(element.HtmlId, outer.TopLeft.X, outer.TopLeft.Y, outer.TopRight.X, outer.TopRight.Y, outer.BottomRight.X, outer.BottomRight.Y, outer.BottomLeft.X, outer.BottomLeft.Y);
+				element.SetStyle("overflow", "hidden"); // overflow: hidden is required here because the clipping can't do its job when it's non-rectangular.
```

Comment:
Can this be done inside the `SetCornerRadius` as well? (if it makes sense)

Reply:
Indeed, that would be interesting.

Reply:
Done!